src/main/java/com/faforever/api/error/GlobalControllerExceptionHandler.java

```diff
@@ -0,0 +1,107 @@
+package com.faforever.api.error;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.InsufficientAuthenticationException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.ValidationException;
+import java.text.MessageFormat;
+import java.util.Arrays;
+
+@ControllerAdvice
+// TODO implement a proper JSON-API error response
+class GlobalControllerExceptionHandler {
+
+  @ExceptionHandler(ConstraintViolationException.class)
+  @ResponseStatus(HttpStatus.UNPROCESSABLE_ENTITY)
+  @ResponseBody
+  public ErrorResponse processValidationException(ConstraintViolationException ex) {
+    final ErrorResponse errorResponse = new ErrorResponse();
+    ex.getConstraintViolations().forEach(constraintViolation -> {
+      errorResponse.addError(new ErrorResult(
+        String.valueOf(HttpStatus.UNPROCESSABLE_ENTITY.value()),
+        String.valueOf(ErrorCode.VALIDATION_FAILED.getCode()),
+        ErrorCode.VALIDATION_FAILED.getTitle(),
+        constraintViolation.getPropertyPath().toString() + " " + constraintViolation.getMessage(),
```

Comment:
Call to `toString()` is not needed since it's a string concatenation (I think IntelliJ highlights this).
Also, I usually prefer `String.format()` over concatenation, and the format `%s: %s`.